ruby/examples/lib/register_user_with_form.rb

```diff
@@ -0,0 +1,39 @@
+require 'active_model'
+
+class RegisterUserWithForm
+  Request = Struct.new(:email, :name, :password)
+  Response = Struct.new(:form)
+
+  class Response::Form
+    include ActiveModel::Model
+
+    attr_reader *Request.members
+    attr_reader :errors
+
+    def initialize(email:, name:, password:, errors:)
+      @email = email
+      @name = name
+      @password = password
+      @errors = errors
+    end
+  end
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
+    @gateway = gateway
+  end
+
+  def register(request)
+    errors = gateway.create_user(request.to_h)
+    form = Response::Form.new(request.to_h.merge(errors: errors))
+    Response.new(form)
+  end
+
+  class Gateway
+    def create_user(attributes)
+      order = Spree::User.create(attributes)
+      order.errors
```

Comment:
I think the fact we see Spree here is muddying things. Think personally we would wrap the Spree validations up into an interface the use case uses... then you'd depend on it as follows

```ruby
spree_adapter = SpreeOrderAdapter.new

use_case = SomeUseCase.new(order_validator: spree_adapter, order_gateway: spree_adapter)
```